src/main/java/net/openhft/chronicle/wire/GenerateMethodReader.java

```diff
@@ -298,16 +297,10 @@ private void handleMethod(Method m, Class<?> anInterface, String instanceFieldNa
             if (parameterType.isPrimitive())
                 fields.append(format("private %s %sarg%d;\n", typeName, m.getName(), i));
             else {
-                fields.append(format("private %s %sarg%dtype = ObjectUtils.implementationToUse(%s.class);\n",
-                        "Class", m.getName(), i, typeName));
-
-                final Class<?> implToUse = ObjectUtils.implementationToUse(parameterType);
-
-                if (ReadMarshallable.class.isAssignableFrom(implToUse)) {
-                    fields.append(format("private %s %sarg%d = ObjectUtils.newInstance(%s.class);\n",
-                            typeName, m.getName(), i, implToUse.getCanonicalName()));
-                } else
-                    fields.append(format("private %s %sarg%d;\n", typeName, m.getName(), i));
+                // this used to use ObjectUtils.implementationToUse to determine the type and mutable
+                // object but this is not appropriate, especially for an interface type - see #253
+                fields.append(format("private %s %sarg%dtype = %s.class;\n", "Class", m.getName(), i, typeName));
```

Comment:
The only place where %sarg%dtype field used is here:
```
            return format("%s = %s.object(checkRecycle(%s), %stype);\n", argumentName, valueInName, argumentName, argumentName);
```
As typename is now determined at the moment of code generation, we can remove the field.

Reply:
Yes great idea - have made that change. Please take a look!

Reply:
Based on our emails I think this is safe to merge - I am keen to deliver to customer - thanks for feedback